### PR TITLE
docs: set max_completion_length=1024 in GRPO quickstart examples

### LINKED ISSUE #5697: Bug: GRPO quickstart `max_completion_length=256` default silently breaks training
### Reproduction

I was trying to get started with trl using the example in the `README.md`

```python
# copy-paste from README, single GPU
from datasets import load_dataset
from trl import GRPOConfig, GRPOTrainer
from trl.rewards import accuracy_reward

dataset = load_dataset("trl-lib/DeepMath-103K", split="train")

training_args = GRPOConfig(
    output_dir="output",
    # all defaults: per_device_train_batch_size=8, num_generations=8, max_completion_length=256
    logging_steps=1,
)

trainer = GRPOTrainer(
    model="Qwen/Qwen2.5-0.5B-Instruct",
    reward_funcs=accuracy_reward,
    train_dataset=dataset,
    args=training_args,
)
trainer.train()
```

**Result:** Almost all completion hits exactly 256 tokens. The reward is 0 at most steps. 

```
Step  loss  grad_norm  reward_mean  clipped_ratio  mean_length
1     0     0          0            1.0            256
2     0     0          0            1.0            256
...
10    0     0          0            1.0            256
```

**Default `max_completion_length=256` causes near-total truncation on the recommended dataset.** The quickstart pairs `Qwen/Qwen2.5-0.5B-Instruct` with `DeepMath-103K`, but the default 256-token limit truncates 97-99% of completions. The model needs ~600-700 tokens to produce a complete math solution, so it is physically cut off before it can output `\boxed{...}` and receive any reward. The model sees zero gradients for almost the entire run initially.

**Fix:** Set `max_completion_length=1024`. Truncation drops from 97-99% to 8%. Reward appears within the first few steps.

## Evidence

I ran 3 training runs (100 steps each, all defaults: `per_device_train_batch_size=8`, `num_generations=8`, A100 80GB), varying only `max_completion_length`:

| `max_completion_length` | Mean Truncation Rate | Mean Reward | Peak GPU Memory |
|---|---|---|---|
| **256 (default)** | **98.4%** | **0.005** | **14.7 GB** |
| **1024** | **14.2%** | **0.114** | **32.4 GB** |
| **2048** | **0.6%** | **0.146** | **73.6 GB** |

### Truncation rate and reward over 100 training steps

Image

Left: at 256 tokens, `completions/clipped_ratio` is pinned at ~100% for the entire run — the model almost never finishes a single answer. At 1024, truncation drops to ~14%. Right: at 256 tokens, reward flatlines at 0. At 1024, reward signal appears immediately. The same model, same dataset, same reward function produces ~23x more reward simply by allowing enough tokens to finish answering. Note the memory cost: going from 256 to 1024 more than doubles peak GPU memory (14.7 GB to 32.4 GB), and 2048 requires 73.6 GB — nearly filling an A100 80GB.

### System Info

- Platform: Linux-6.1.0-45-cloud-amd64-x86_64-with-glibc2.36
- Python version: 3.11.2
- TRL version: 1.1.0
- PyTorch version: 2.11.0
- accelerator(s): NVIDIA A100-SXM4-80GB
- Transformers version: 5.5.4
- Accelerate version: 1.13.0
- Datasets version: 4.8.4
- HF Hub version: 1.10.2

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

### PR DESCRIPTION
**Pull request by dhruvnigam93**
# What does this PR do?

The default `max_completion_length=256` truncates 97-99% of completions on DeepMath-103K before the model can emit `\boxed{...}`, so `accuracy_reward` scores everything 0. Training runs silently with zero reward, zero loss, and zero gradient updates for most steps without errors. A user following the quickstart will watch flat metrics for hours with no indication that anything is wrong.

This PR sets `max_completion_length=1024` in all three quickstart GRPO examples (README, `quickstart.md`, `grpo_trainer.md`). At 1024, truncation drops to ~14%, and the reward signal appears within the first few steps. Verified on actual runs on a single A100. Details in #5697. 

Also adds single-GPU timing and memory guidance (A100 80GB: ~5 days, ~32 GB peak). Quickstart users are most likely to run on a single GPU, so setting realistic hardware expectations upfront prevents wasted compute.

Fixes #5697

## Before submitting

- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request), Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## AI writing disclosure

- [ ] No AI usage: the PR was written entirely by a human.
- [x] AI-assisted: some parts were suggested or improved by AI, but the PR was written and reviewed by a human.
- [ ] AI-generated: the PR was mostly or fully generated by an AI tool.

## Who can review?

@qgallouedec

---

> [!NOTE]
> **Low Risk**
> Doc-only changes updating GRPO example configuration; no runtime/library behavior is modified.
> 
> **Overview**
> Updates the GRPO quickstart examples in `README.md`, `docs/source/quickstart.md`, and `docs/source/grpo_trainer.md` to construct a `GRPOConfig` and pass it via `args`, explicitly setting `max_completion_length=1024` (and `output_dir`).
> 
> Adds an additional note in the GRPO trainer docs with expected single-GPU (A100 80GB) training time and peak memory usage alongside the existing 8-GPU estimate.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d4d2f04c3f2ce254c164fe71308580245c427b81. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by qgallouedec** [comment]
Thanks @dhruvnigam93, this is a real footgun and the evidence in #5697 is convincing. However, the goal is more to give the absolute minimal code to start with. So it's rather change the dataset (maybe find an easier one -> shorter completion), than adding an optional argument (`args`).

I'll close the PR, but feel free to open a new one suggesting a new dataset, or it can even be another model that generates shorter completions.

**Comment by dhruvnigam93** [comment]
I tested whether **any** small Qwen model can produce GSM8K completions with fewer than 256 tokens. We tested across 3 model families, 3 sizes, and 2 architecture generations — including a math-specialized model and Qwen3's non-thinking mode. **None of them can reliably fit chain-of-thought reasoning within 256 tokens**

I think we need to question if 256 is the right default here. GRPO is designed for multi-step reasoning tasks. Multi-step reasoning inherently produces completions longer than 256 tokens. The default `max_completion_length=256` is incompatible with the algorithm's intended use case.

GSM8K is the simplest widely-used math reasoning benchmark (grade-school arithmetic). If any reasoning task should fit in 256 tokens, it's this one.

SO I tested across:
- **3 model sizes** (0.5B, 0.6B, 1.5B)
- **2 model generations** (Qwen 2.5, Qwen 3)
- **2 specializations** (general instruct, math-specialized)
- **Both thinking and non-thinking modes** (Qwen3's `enable_thinking=False`)

**No model produces GSM8K completions reliably under 256 tokens.**


## Summary of all models tested

| Model | Size | Generation | Specialization | Median tokens | Truncation @256 
|-------|:---:|:---:|:---:|:---:|:---:|
| Qwen2.5-0.5B-Instruct | 0.5B | 2.5 | General | ~274 | ~57% 
| **Qwen2.5-Math-1.5B-Instruct** | 1.5B | 2.5 | **Math** | ~315 | ~80% 
| **Qwen3-0.6B (no_think)** | 0.6B | 3.0 | General | ~248 | ~50% 

It seems that `max_completion_length=256` is not feasible even for the simplest problem set. I will open an issue for this, and we can follow up there. 

Tangentially, I feel there should be a warning when all completions are truncated, resulting in no rewards or training. This could help users debug more quickly. Raise the PR here #6220
